vllm/config.py

```diff
@@ -296,6 +296,9 @@ def _verify_args(self) -> None:
             raise ValueError(
                 "GPU memory utilization must be less than 1.0. Got "
                 f"{self.gpu_memory_utilization}.")
+        if self.prefix_pool_max_capacity is not None and self.prefix_pool_max_capacity <= 0:
```

Comment:
How about profiling the maximum capacity by current max gpu memory if user doesn't set this parameter? And raising error when the server cannot afford larger maximum capacity.

Reply:
@esmeetu Last commit just introduced has changed the `PrefixPool` implementation to use number of blocks instead of number of prefixes.